app/feature/feature-help-center/src/main/kotlin/com/hedvig/android/feature/help/center/data/GetQuickLinksUseCase.kt

```diff
@@ -90,13 +89,26 @@ internal class GetQuickLinksUseCase(
           }
         }
 
+      contracts
+        .takeIf { featureManager.isFeatureEnabled(Feature.TERMINATION_FLOW).first() }
+        ?.let {
+          add(
+            QuickAction.QuickLink(
+              quickLinkDestination = QuickLinkDestination.QuickLinkTermination,
+              titleRes = R.string.HC_QUICK_ACTIONS_CANCELLATION_TITLE,
+              hintTextRes = R.string.HC_QUICK_ACTIONS_CANCELLATION_SUBTITLE,
+              displayName = null,
+            ),
+          )
+        }
```

Comment:
The null `displayName` here, which part of the UI is that driving?
Also I wonder, besides hiding when the feature flag is off, do we also need to hide this quick action for people who do not have any active insurance right now? Like the same check we have to do in the termination flow itself which checks if they can select any of their insurances to terminate?

If the answer is yes, it may be worth it extracting that piece of logic in some sort of ":data-termination" module which we can use here, and then the termination flow screen can also share that same piece of logic. 

What do you think, am I missing something here?

Reply:
displayName is shown in the select dialog, if we have multiple insurances to edit co-ensured. we don't need it for cancellation.
Agree with the separate :data-termination module, will do that!

Reply:
Sounds like an opportunity to perhaps make this easier to use by having those two cases be distinct classes inside some sort of sealed hierarchy perhaps?
So that one can't misuse this API by accident. Or if they provide a displayName and expect it to show, but it just doesn't for this case?

Reply:
divided them now:)